src/metrics/mod.rs

```diff
@@ -367,7 +374,7 @@ pub async fn middleware(
                 conn_reqs = conn_req_count,
                 cache_status = cache_status_str,
                 cache_bypass_reason = cache_bypass_reason_str,
-                backend = req_meta.backend,
+                upstream = req_meta.backend,
```

Comment:
Let's unify maybe and use `backend_host` everywhere since it was added.

Reply:
Sure. In addition, IMO [BNRequestMetadata](https://github.com/dfinity/ic-gateway/blob/424387d1262cda897a19df765815f4cd331454fc/src/routing/ic/mod.rs#L33) should be modified, as keeping backend's hostname and port in one string is not convenient for parsing.

Reply:
But then you'll have to parse it anyway before putting into the struct, what's the difference? Also for known ports (80/443) the port is anyway not added there, though we have to anyway account that it might be there.

Reply:
If we'll have more use cases for this backend - let's then move the parsing to the source, for now we have only one place where it's used (here in metrics) and it makes no difference. And the port (potentially) might be needed in certain cases.